lib/Ast.ml

```diff
@@ -1801,22 +1721,27 @@ end = struct
       binary operators, otherwise returns [None] if [ctx] has no precedence
       or [Some None] if [ctx] does but [ast] does not. *)
   let ambig_prec ({ast; _} as xast) =
-    prec_ctx xast
-    >>| fun (prec_ctx, which_child) ->
-    prec_ast ast
-    >>| fun prec_ast ->
-    match compare_prec prec_ctx prec_ast with
-    | 0 ->
-        (* which child and associativity match: no parens *)
-        (* which child and assoc conflict: add parens *)
-        equal_assoc which_child Non
-        || not (equal_assoc (assoc_of_prec prec_ast) which_child)
-    | cmp when cmp < 0 ->
-        (* ast higher precedence than context: no parens *)
-        false
-    | _ (* > 0 *) ->
-        (* context higher prec than ast: add parens *)
-        true
+    match prec_ctx xast with
+    | Some (prec_ctx, which_child) -> (
+      match prec_ast ast with
+      | Some prec_ast ->
+          let ambiguous =
+            match Prec.compare prec_ctx prec_ast with
+            | 0 ->
+                (* which child and associativity match: no parens *)
+                (* which child and assoc conflict: add parens *)
+                Assoc.equal which_child Non
+                || not (Assoc.equal (Assoc.of_prec prec_ast) which_child)
+            | cmp when cmp < 0 ->
```

Comment:
This and the next case can be written `| cmp -> cmp >= 0` (and one of the two comment can be dropped).